src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -681,6 +681,73 @@ private static BoundNode Rewrite(ImmutableDictionary<BoundExpression, (Nullabili
             }
         }
 
+        /// <summary>
+        /// Check that slots for member symbols indicate members of the enclosing slot's type.
+        /// </summary>
+        [Conditional("DEBUG")]
+        private void ValidateLastSlot()
+        {
+            int i = nextVariableSlot - 1;
+            if (i <= 1)
+                return;
+
+            var variable = variableBySlot[i].Symbol;
+            var containingSlot = variableBySlot[i].ContainingSlot;
+            if (containingSlot < 1)
+                return;
+
+            var containingSlotType = variableBySlot[containingSlot].Symbol.GetTypeOrReturnType().Type;
+            Debug.Assert(IsMember(variable, containingSlotType));
+        }
+
+        static bool IsMember(Symbol member, TypeSymbol possibleContainer)
+        {
+            TypeSymbol memberContainer = member.ContainingType;
+            if (memberContainer.Equals(possibleContainer, TypeCompareKind.AllIgnoreOptions))
```

Comment:
>memberContainer.Equals(possibleContainer, TypeCompareKind.AllIgnoreOptions) [](start = 16, length = 75)

Consider extracting a local function since the same check is made several times.